php: update , bugfix , enhancement to the whole php infrastructure

BjarneDMat opened the pull request:
## Description

php: upgrade , bugfix , enhancement to the whole php infrastructure
see: [https://macports.mathiesen.info/macports/](https://macports.mathiesen.info/macports/) for scripts

Presently, I'm fully content with just having my php84 branch w/ all of the necessary changes for getting the php infrastructure up-to-date collected in one, single place. Until someone w/ commit authority turns up w/ some constructive feedback as to how, they want me to proceed, I'm not changing anything at all. I've asked the maintainers several times, how they want to proceed, without getting any feedback at all 👎🏻. Actually, for the folks interested in the php infrastructure, it's an advantage to have everything collected in one place. I've given instructions on how, one can integrate my php84 branch, into one's MacPorts installation on the macports users mailing list. And you can find the relevant scripts by following the above link. When I update my php84 branch, I'll continue to post these on both the users & developer mailing lists to keep everyone up-to-date.

Presently, I'm (primarily) updating PR28140 whenever php releases new versions. Between php releases, I'm working on the php-* extensions; and then update the whole PR28140 in one go with a list of what's changed published both in the PR28140 and to the mailing lists. That also means, that I've always got a few things in the pipeline, that hasn't yet been force-pushed.
```
+------------------------------------------+
| Accumulated updates etc as of 2025-05-09 |
+------------------------------------------+
php8.{1..5}            : updated to 8.1.34, 8.2.31, 8.3.31, 8.4.21, 8.5.6
autoconf               : C23 [-Wdeprecated-non-prototype] fixes

php-pcntl              : php{74,80} fix [-Wincompatible-function-pointer-types] in ext/pcntl/pcntl.c
php-oracle             : php74 fix for [-Wincompatible-function-pointer-types] in ext/pdo_oci/oci_statement.c
php-odbc               : php{74,80} fix for [-Wincompatible-function-pointer-types] in ext/pdo_odbc/odbc_stmt.c
php-intl               : php{74,80} fix K&R function declarations in collator_is_numeric.c
php-gd                 : php{74,80} patches to gd.c

php-xapian             : update to 2.0.0 ; added php8[45] , perl5.40 , python314
php-sqlsrv             : update to 5.13.1 for php >= 8.3
msodbcsql              : update @18.6.2.1 ; openssl3
php-oracle             : new port @1.2.0
```

### Instructions
In order to clone & get this to work on your own computer, this is my recommended procedure:

[https://macports.mathiesen.info/macports/](https://macports.mathiesen.info/macports/)
from gitRepos, you'll need :
1. gitFunctions.include
2. php84Git

I order to get it to work, you'll need a directory to work in - eg, **MacPorts** so :

start by installing bash, curl & git:
`port -cuN install bash curl git`
and use MacPorts `bash` as your bash in Terminal

Then:
```
declare mathiesen='https://macports.mathiesen.info/macports/gitRepos'
mkdir -p MacPorts/gitRepos ; cd MacPorts/gitRepos
curl -O "${mathiesen}/gitFunctions.include"
curl -O "${mathiesen}/php84Git"
chmod +x php84Git
sed -i '' -E -e '/^case/s/skip/update/' gitFunctions.include
cd ..
./gitRepos/php84Git init
```
then, you'll have to edit (as root) `/opt/local/etc/macports/sources.conf`
and insert an absolute path to your cloned `BjarneDMat:php84` - in my case :
`file:///Users/macuser/MacPorts/ports/php84      [nosync]`

you can test, if you've gotten everything right, by executing :
`port file php84`

Subsequent updates can then simply be executed by :
`./gitRepos/php84Git`
as `update` now is the default

###### Closes these duplicates

- ~~#30533.~~
- ~~#29866~~
- ~~#30146~~
- ~~#30124~~
- ~~#30123~~

###### Closes these tickets

- ~~[73529](https://trac.macports.org/ticket/73529) [Port Abandoned] php83~~
- ~~[72208](https://trac.macports.org/ticket/72208) php81 php82 php83 php84 update~~
- ~~[71999](https://trac.macports.org/ticket/71999) Release stable version of PHP 8.4 and add some missing extensions for it~~
- ~~[71302](https://trac.macports.org/ticket/71302) php83-openswoole: port request~~
- ~~[73248](https://trac.macports.org/ticket/73248) php84-zip: Is not available in the php84 ports~~


###### Type(s)
- [x] upgrade
- [x] bugfix
- [x] enhancement
- [ ] security fix

###### Tested on
```
Model Identifier: MacPro5,1                 Model Identifier: Macmini6,1
macOS 15.7.4 24G517 x86_64                  macOS 10.15.8 19H2036 x86_64
Xcode 26.3 17C529                           Xcode 12.4 12D4e
Command Line Tools 26.2.0.0.1.1764812424    Command Line Tools 12.4.0.0.1.1610135815

Model Identifier: Mac14,3
macOS 26.2 25C56 arm64
Xcode not installed
Command Line Tools 26.2.0.0.1.1764812424
```
I'm only explicitly testing for  `${php.branch} >= 7.4` .
I regard every `${php.branch} < 7.4` as being ☠️🪦⚰️⚱️👻

BjarneDM committed:
On branch master
Your branch is up to date with 'origin/master'.

BjarneDM committed:
php8.{1..4}: updated to 8.1.33, 8.2.29, 8.3.28, 8.4.15

This collect every version update before the release of 8.5.0

Changes to be committed:
	modified:   _resources/port1.0/group/php-1.1.tcl
		    set default php to 8.4
		    fix for lint : Error: Portfile parent directory lang does not match primary category php
	modified:   lang/php/Portfile
		    fix for lint : Error: Portfile parent directory lang does not match primary category php
	modified:   php/php-APCu/Portfile
	modified:   php/php-igbinary/Portfile
	modified:   php/php-imagick/Portfile    <- building fails
	modified:   php/php-lzf/Portfile
	modified:   php/php-mailparse/Portfile
	modified:   php/php-mcrypt/Portfile     <- building fails
	modified:   php/php-memcached/Portfile
	modified:   php/php-mongodb/Portfile
	modified:   php/php-raphf/Portfile
	modified:   php/php-redis/Portfile
	modified:   php/php-timezonedb/Portfile
	modified:   php/php-vld/Portfile        <- configure fails
	modified:   php/php-xdebug/Portfile
	modified:   php/php-yaml/Portfile
	modified:   php/php-zip/Portfile

	deleted:    lang/php/files/patch-php83-ext-hash-xxhash.diff <- fixed in 8.3.28

Inserted placeholder for php85

BjarneDM committed:
php-imap: for php >= 8.3 moved to separate Portfile

Changes to be committed:
	modified:   lang/php/Portfile
	new file:   php/php-imap/Portfile

BjarneDM committed:
php-pspell: for php >= 8.1 moved to separate Portfile

Changes to be committed:
	modified:   lang/php/Portfile
	new file:   php/php-pspell/Portfile

BjarneDM committed:
libxml2 is now @2.13.5_2

Changes to be committed:
	modified:   lang/php/Portfile
	new file:   lang/php/files/ext-xml-tests-gh14834.phpt
	deleted:    lang/php/files/patch-php52-libxml2-2.12.diff
	deleted:    lang/php/files/patch-php53-libxml2-2.12.diff
	deleted:    lang/php/files/patch-php54-libxml2-2.12.diff
	deleted:    lang/php/files/patch-php55-libxml2-2.12.diff
	deleted:    lang/php/files/patch-php56-libxml2-2.12.diff
	deleted:    lang/php/files/patch-php70-libxml2-2.12.diff
	deleted:    lang/php/files/patch-php71-libxml2-2.12.diff
	deleted:    lang/php/files/patch-php72-libxml2-2.12.diff
	deleted:    lang/php/files/patch-php73-libxml2-2.12.diff
	new file:   lang/php/files/patch-php74-ext-xml-compat.c.diff
	new file:   lang/php/files/patch-php74-libxml.diff
	new file:   lang/php/files/patch-php74-libxml.diff.fails
	new file:   lang/php/files/patch-php74-libxml.diff.orig
	new file:   lang/php/files/patch-php74-libxml.diff.works
	deleted:    lang/php/files/patch-php74-libxml2-2.12.diff
	new file:   lang/php/files/patch-php80-ext-xml-compat.c.diff
	new file:   lang/php/files/patch-php80-libxml.diff
	new file:   lang/php/files/patch-php80-libxml.diff.fails
	new file:   lang/php/files/patch-php80-libxml.diff.orig
	renamed:    lang/php/files/patch-php80-libxml2-2.12.diff -> lang/php/files/patch-php80-libxml.diff.works
	deleted:    lang/php/files/patch-php81-libxml2-2.12.diff

BjarneDM committed:
xxhash patches for php81, php82, php83

php83 fixed in 8.3.29
php82 fixed in 8.2.30
php81 fixed in 8.1.34

Changes to be committed:
	modified:   lang/php/Portfile
	new files:  lang/php/files/patch-{phpversion}-ext-hash-xxhash.diff

BjarneDM committed:
php-imagick: update to 3.8.1

Changes to be committed:
	modified:   php/php-gmagick/Portfile
	modified:   php/php-imagick/Portfile

BjarneDM committed:
php-mcrypt: updated to 1.0.9

BjarneDM committed:
php-vld: update to 0.19.1

Changes to be committed:
	modified:   php/php-vld/Portfile

BjarneDM committed:
php-*: updating externsions to php84

The following extensions presently fail to configure -or- build:
                     @8.2.27 @8.3.16 @8.4.3
php-mysql_xdevapi                       x
php-rar                 x       x       x
php-openswoole                  x       x
php-swoole                      x       x
php-vld                                 x
php-xapian              x       x       x
php-yaf                                 x

On branch php84

Changes to be committed:
	modified:   devel/xapian-core/Portfile
	modified:   php/php-amqp/Portfile
	modified:   php/php-dbase/Portfile
	modified:   php/php-event/Portfile
	modified:   php/php-gearman/Portfile
	modified:   php/php-geoip/Portfile
	modified:   php/php-gmagick/Portfile
	modified:   php/php-jsmin/Portfile
	modified:   php/php-maxminddb/Portfile
	modified:   php/php-memcache/Portfile
	modified:   php/php-meminfo/Portfile
	modified:   php/php-mysql_xdevapi/Portfile
	modified:   php/php-oauth/Portfile
	modified:   php/php-openswoole/Portfile
	modified:   php/php-pcov/Portfile
	modified:   php/php-psr/Portfile
	modified:   php/php-rar/Portfile
	modified:   php/php-rrd/Portfile
	modified:   php/php-scrypt/Portfile
	modified:   php/php-solr/Portfile
	modified:   php/php-sqlsrv/Portfile
	modified:   php/php-ssh2/Portfile
	modified:   php/php-stomp/Portfile
	modified:   php/php-svm/Portfile
	modified:   php/php-swoole/Portfile
	modified:   php/php-uploadprogress/Portfile
	modified:   php/php-uuid/Portfile
	modified:   php/php-xhprof/Portfile
	modified:   php/php-xmlrpc/Portfile
	modified:   php/php-yaf/Portfile
	modified:   php/php-yaz/Portfile
	modified:   php/php-zstd/Portfile

BjarneDM committed:
php-postgresql: added variant postgresql{17,18}

Changes to be committed:
	modified:   lang/php/Portfile

BjarneDM committed:
php-mysql : updated mariadb based variant; removed obsolete stuff

replaced the outdated mariadb variant with newer LTS variants based upon
	mariadb10 -> port:mariadb-10.11
	mariadb11 -> port:mariadb-11.8
	mariadb12 -> port:mariadb-12.3

percona has been removed from MacPorts

BjarneDM committed:
php74: implicit ints problem

1st attempt at fixing this
https://trac.macports.org/ticket/70468
php74 @7.4.33: Undefined symbols _res_9_dn_expand, _res_9_dn_skipname, _res_9_init, _res_9_search,

by implementing the changes from
https://github.com/php/php-src/commit/aeaab8e
Port various autoconf bits to C99 compilers

Changes to be committed:
	modified:   lang/php/Portfile
	new file:   lang/php/files/patch-php73-implicit.diff <- placeholder
	new file:   lang/php/files/patch-php74-implicit.diff

BjarneDM committed:
php8.{1..5}: updated to 8.1.34, 8.2.30, 8.3.30, 8.4.19, 8.5.4

Commit History:
libxml2: 2.13.9_0 < 2.15.2_0
	SAPI subports RevBumped : apache2handler cgi fpm ${php.rootname}
	php-solr php-xmlrpc RevBumped

php8.{1..5}: updated to 8.1.34, 8.2.30, 8.3.29, 8.4.16, 8.5.1
        deleted:    lang/php/files/patch-php81-ext-hash-xxhash.diff
        deleted:    lang/php/files/patch-php82-ext-hash-xxhash.diff
        deleted:    lang/php/files/patch-php85-mysqlnd-connection.c.diff
        deleted     lang/php/files/patch-php8[12345]-zend_string_equal_val.diff

php: libxml2 2.13.9_0 < 2.15.2_0

konwn-fail functionalty based upon branch

ryandesign commented:
Obviously a commit bomb like this is impossible for me to review, not to mention that is has conflicts do it can't be merged.

BjarneDMat commented:
what do you want me to do with it ?
I've tried asking you on the mailing lists what the best way would be
The conflict is because of the proposed changes to php-mysql and the mariadb variants, which are also in https://github.com/macports/macports-ports/pull/28046

BjarneDMat commented:
there're also some problems that are still un-resolved that I need some help with

BjarneDMat commented:
> In this PR, CI-related files are deleted, which is causing conflicts with the master branch. It was probably done to prevent GitHub Actions from running on the user's cloned repository, but this should be handled by disabling Actions in the repository settings.

@rsky I must confess, I've absolutely no idea at all what you're talking about ... I ought to have done exactly the same things as I've done w/ all other PRs I've sumbittet. I know, I've done nothing to explicitly disable GitHub Actions on my own repo.

What can I do to fix this issue ?

miken32 commented:
@BjarneDM I'm guessing this PR should be closed and the changes split up into smaller chunks if you want to get any traction on them. An 8.4 release, new ports for imap and pspell, backported XML fixes for unsupported versions, etc. – all these should be separate PRs IMO. Ideally you'd get some guidance on this from project owners but that doesn't seem to be happening here.

BjarneDM committed:
php85: new subport branch 8.5.0

On branch php84
Changes to be committed:

Core 8.5.0
        modified:   _resources/port1.0/group/php-1.1.tcl
	modified:   lang/php/Portfile
	new file:   lang/php/files/patch-php85-atomic.diff
	new file:   lang/php/files/patch-php85-iODBC.diff
	new file:   lang/php/files/patch-php85-sapi-fpm-php-fpm.conf.in.diff
	new file:   lang/php/files/patch-php85-scripts-php-config.in.diff
	new file:   lang/php/files/patch-php85-unixODBC.diff
	new file:   lang/php/files/patch-php85-zend_string_equal_val.diff
        new file:   lang/php/files/patch-php85-mysqlnd-connection.c.diff
	new file:   lang/php/files/php85

find php -name Portfile \
| xargs -n 1 sed -i '' -E -e 's/^(php.branches .*8.4)$/\1 8.5/'

https://github.com/remicollet/remirepo/issues/307

Extensions that have been patched:
	php-amqp       : patched for php >= 8.5
	php-gmagick    : patched for php >= 8.5
	php-gearman    : patched for php >= 8.5
	php-jsmin      : patch for zend_smart_string
	php-memcache   : patches for zend_smart_string
	php-openswoole : patched for php >= 8.1
	php-svm        : patched for php >= 8.5
	php-stomp      : patched for php >= 8.5
	php-swoole     : patched for php >= 8.1

Extensions that presently fail to install:
	modified:   php/php-rar/Portfile
	modified:   php/php-rrd/Portfile

Extensioan that presently install
	modified:   devel/xapian-core/Portfile
	modified:   php/php-APCu/Portfile
	modified:   php/php-dbase/Portfile
	modified:   php/php-event/Portfile
	modified:   php/php-geoip/Portfile
	modified:   php/php-igbinary/Portfile
	modified:   php/php-imagick/Portfile
	modified:   php/php-imap/Portfile
	modified:   php/php-lzf/Portfile
	modified:   php/php-mailparse/Portfile
	modified:   php/php-maxminddb/Portfile
	modified:   php/php-mcrypt/Portfile
	modified:   php/php-meminfo/Portfile
	modified:   php/php-memcached/Portfile
	modified:   php/php-mongodb/Portfile
	modified:   php/php-mysql_xdevapi/Portfile
	modified:   php/php-oauth/Portfile
	modified:   php/php-pcov/Portfile
	modified:   php/php-pspell/Portfile
	modified:   php/php-psr/Portfile
	modified:   php/php-raphf/Portfile
	modified:   php/php-redis/Portfile
	modified:   php/php-scrypt/Portfile
	modified:   php/php-solr/Portfile
	modified:   php/php-sqlsrv/Portfile
	modified:   php/php-ssh2/Portfile
	modified:   php/php-timezonedb/Portfile
	modified:   php/php-uploadprogress/Portfile
	modified:   php/php-uuid/Portfile
	modified:   php/php-vld/Portfile
	modified:   php/php-xdebug/Portfile
	modified:   php/php-xhprof/Portfile
	modified:   php/php-xmlrpc/Portfile
	modified:   php/php-yaf/Portfile
	modified:   php/php-yaz/Portfile
	modified:   php/php-yaml/Portfile
	modified:   php/php-zip/Portfile
	modified:   php/php-zstd/Portfile

beryllium commented:
Thanks for your work on this Bjarne. I hope someone in the project can provide more clear guidance on how to move forward with this; as a PHP dev, I am sorely missing things like php84-memcached & it is impacting my hobby workflow. (Less impactful on my work workflow, since we're not on 8.4 yet)

I don't have much portfile experience, but please let me know if I can help.

BjarneDM committed:
php-zip: update to 1.22.8

BjarneDM committed:
php-xdebug: update to 3.5.0, 3.4.7 ; livecheck

BjarneDM committed:
php-uuid: update to 1.3.0

BjarneDM committed:
php-APCu: update to 5.1.28

BjarneDM committed:
php-yaf: update to 3.3.7

BjarneDM committed:
php-timezonedb: update to 2026.1

BjarneDM committed:
php-solr: update to 2.9.1_1

BjarneDM committed:
php-redis: update to 6.3.0

BjarneDM committed:
php-maxminddb: update to 1.13.1

osbre commented:
For users: as a temporary solution, I went with Herd's PHP installer:

```php
/bin/bash -c "$(curl -fsSL https://php.new/install/mac/8.4)"
```
https://laravel.com/docs/12.x#installing-php

hostep commented:
I've switched to https://github.com/shivammathur/homebrew-php & https://github.com/shivammathur/homebrew-extensions by now, they support PHP 5.6 all the way to PHP 8.4 on the latest macOS version.

Always been very happy with macports, but if they no longer find energy/time to support PHP (which is totally fine, it's open source after all) it's time to switch to another way of installing it on macOS.

GarrettAlbright commented:
Glad to see there's still some discussion and progress on this. PHP 8.5 is two months away so we definitely need to get this unstuck. If there's anything I can do to help please let me know. MacPorts has been the part of my workflow for the better part of two decades now and I'd rather not have to switch and relearn how to use everything, but that same workflow is going to need current versions of PHP available soon. Thanks to all for their work.

beryllium commented:
Thanks for keeping this up to date! Seems like the concerns from the other maintainers are mostly addressed now - maybe they'll consider moving it out of Draft status & getting it merged.

BjarneDM committed:
php-mailparse: update to 3.1.9

BjarneDM committed:
php-zstd: update to 0.15.2

BjarneDM committed:
php-yaml: update to 2.3.0

BjarneDM committed:
php-oauth: update to 2.0.10

BjarneDM committed:
php-amqp: update to 2.2.0

BjarneDM committed:
php-gearman: patched for php >= 8.5

BjarneDM committed:
php-igbinary: update to 3.2.17RC1

BjarneDM committed:
php-gmagick: patched for php >= 8.5

BjarneDM committed:
php-jsmin: patch for zend_smart_string

BjarneDM committed:
php-memcache: patches for zend_smart_string

BjarneDM committed:
php-memcached: update to 3.4.0

BjarneDM committed:
php-swoole: update to 6.2.0 >= php82; update to 6.1.7, 5.1.8

incorporates work from https://github.com/macports/macports-ports/pull/30146

BjarneDM committed:
php-stomp: patched for php >= 8.5

BjarneDM committed:
php-svm: patched for php >= 8.5

BjarneDM committed:
php-openssl: fix for Dependency path specified multiple times in depends_build

BjarneDM committed:
php-* extensions: fix for categories

change 'categories' to 'categories-append'
remove 'php' from the 'categories-append' list

Changes to be committed:
	modified:   php/php-APC/Portfile
	modified:   php/php-FreeImage/Portfile
	modified:   php/php-Judy/Portfile
	modified:   php/php-SPL_Types/Portfile
	modified:   php/php-amf/Portfile
	modified:   php/php-bbcode/Portfile
	modified:   php/php-big_int/Portfile
	modified:   php/php-cairo/Portfile
	modified:   php/php-cairo_wrapper/Portfile
	modified:   php/php-colorer/Portfile
	modified:   php/php-crack/Portfile
	modified:   php/php-eaccelerator/Portfile
	modified:   php/php-esmtp/Portfile
	modified:   php/php-gdchart/Portfile
	modified:   php/php-geoip/Portfile
	modified:   php/php-hidef/Portfile
	modified:   php/php-html_parse/Portfile
	modified:   php/php-http/Portfile
	modified:   php/php-http2/Portfile
	modified:   php/php-imlib2/Portfile
	modified:   php/php-inclued/Portfile
	modified:   php/php-lzf/Portfile
	modified:   php/php-mailparse/Portfile
	modified:   php/php-markdown/Portfile
	modified:   php/php-maxminddb/Portfile
	modified:   php/php-mongo/Portfile
	modified:   php/php-peb/Portfile
	modified:   php/php-phalcon/Portfile
	modified:   php/php-pop3/Portfile
	modified:   php/php-rar/Portfile
	modified:   php/php-timezonedb/Portfile
	modified:   php/php-uploadprogress/Portfile
	modified:   php/php-vld/Portfile
	modified:   php/php-xrange/Portfile
	modified:   php/php-xslcache/Portfile

beryllium commented on php/php-xdebug/Portfile:
```diff
@@ -10,18 +10,18 @@ maintainers             {ryandesign @ryandesign} openmaintainer
+php.pecl                yes
 php.extensions.zend     xdebug
 
 if {[vercmp ${php.branch} >= 8.0]} {
```
That sounds safest to me, I wouldn't want to fetch alpha code without people's direct approval.

beryllium replied:
I did try to address the issues with `php-rar` and `php-solr`, but those seem to be suffering from actual build issues (perhaps related to `long` or exceptions?) Unsure how to proceed with those ones. I've compiled from source before, but not for some time & haven't encountered such issues.

I might be able to get some feedback from PHP devs in the next few days, but odds are that this is a quirk of macOS/macports, rather than an issue with the code. php-rar compiles fine for 8.5 on Linux, for example.

BjarneDMat replied:
I've gone with 3.5.0alpha3 for now -- the devs are taking too much time releasing the GA

BjarneDM committed:
php-mongodb: update to 2.2.1 & 1.20.1

ArtKoKo commented:
@ryandesign  Should each port be split into a separate GitHub branch to facilitate revision?

BjarneDM committed:
php-midgard2: PortGroup obsolete 1.0

BjarneDM committed:
php-phalcon: updated {,2,3,4} ; new {5}

Changes to be committed:
	modified:   php/php-phalcon/Portfile
	modified:   php/php-phalcon2/Portfile
	modified:   php/php-phalcon3/Portfile
	modified:   php/php-phalcon4/Portfile

	new file:   php/php-phalcon5/Portfile
	new file:   php/php-phalcon5/files/patch_phalcon.zep.c.diff

BjarneDM committed:
php-taint: removed dependency on git

The git dependeny led to a livecheck error

BjarneDM committed:
php-unit: PortGroup obsolete 1.0 for all ports belonging to php-unit

These are all presently installed using composer

Changes to be committed:
	modified:   php/php-code-coverage/Portfile
	modified:   php/php-file-iterator/Portfile
	modified:   php/php-text-template/Portfile
	modified:   php/php-timer/Portfile
	modified:   php/php-token-stream/Portfile
	modified:   php/php-unit-mock-objects/Portfile
	modified:   php/php-unit/Portfile

BjarneDM committed:
php-raphf: update to 2.0.2

BjarneDM committed:
php5-*: PortGroup obsolete 1.0

Changes to be committed:
	modified:   php/php5-simpletest/Portfile
	modified:   php/php5-unit-db/Portfile
	modified:   php/php5-unit-selenium/Portfile

BjarneDM committed:
phpsh: PortGroup obsolete 1.0

Changes to be committed:
	modified:   php/phpsh/Portfile

BjarneDM committed:
php-*: add BjarneDMat to maintainers

Changes to be committed:
	modified:   php/php-APC/Portfile
	modified:   php/php-FreeImage/Portfile
	modified:   php/php-Judy/Portfile
	modified:   php/php-SPL_Types/Portfile
	modified:   php/php-Twig/Portfile
	modified:   php/php-amf/Portfile
	modified:   php/php-amqp/Portfile
	modified:   php/php-bbcode/Portfile
	modified:   php/php-big_int/Portfile
	modified:   php/php-cairo/Portfile
	modified:   php/php-cairo_wrapper/Portfile
	modified:   php/php-code-coverage/Portfile
	modified:   php/php-colorer/Portfile
	modified:   php/php-crack/Portfile
	modified:   php/php-dbase/Portfile
	modified:   php/php-eaccelerator/Portfile
	modified:   php/php-esmtp/Portfile
	modified:   php/php-event/Portfile
	modified:   php/php-excel/Portfile
	modified:   php/php-gdchart/Portfile
	modified:   php/php-gearman/Portfile
	modified:   php/php-geoip/Portfile
	modified:   php/php-gmagick/Portfile
	modified:   php/php-gtk/Portfile
	modified:   php/php-hidef/Portfile
	modified:   php/php-html_parse/Portfile
	modified:   php/php-htscanner/Portfile
	modified:   php/php-http/Portfile
	modified:   php/php-http2/Portfile
	modified:   php/php-igbinary/Portfile
	modified:   php/php-imagick/Portfile
	modified:   php/php-imlib2/Portfile
	modified:   php/php-inclued/Portfile
	modified:   php/php-jsmin/Portfile
	modified:   php/php-libev/Portfile
	modified:   php/php-libevent/Portfile
	modified:   php/php-lzf/Portfile
	modified:   php/php-magickwand/Portfile
	modified:   php/php-mailparse/Portfile
	modified:   php/php-markdown/Portfile
	modified:   php/php-maxminddb/Portfile
	modified:   php/php-mcrypt/Portfile
	modified:   php/php-memcache/Portfile
	modified:   php/php-memcached/Portfile
	modified:   php/php-meminfo/Portfile
	modified:   php/php-mongo/Portfile
	modified:   php/php-mongodb/Portfile
	modified:   php/php-mysql_xdevapi/Portfile
	modified:   php/php-oauth/Portfile
	modified:   php/php-opcache/Portfile
	modified:   php/php-openswoole/Portfile
	modified:   php/php-pcov/Portfile
	modified:   php/php-pdflib/Portfile
	modified:   php/php-peb/Portfile
	modified:   php/php-phalcon/Portfile
	modified:   php/php-phalcon2/Portfile
	modified:   php/php-phalcon3/Portfile
	modified:   php/php-phalcon4/Portfile
	modified:   php/php-pop3/Portfile
	modified:   php/php-propro/Portfile
	modified:   php/php-psr/Portfile
	modified:   php/php-raphf/Portfile
	modified:   php/php-rar/Portfile
	modified:   php/php-redis/Portfile
	modified:   php/php-rrd/Portfile
	modified:   php/php-scrypt/Portfile
	modified:   php/php-solr/Portfile
	modified:   php/php-sphinx/Portfile
	modified:   php/php-sqlsrv/Portfile
	modified:   php/php-ssh2/Portfile
	modified:   php/php-stomp/Portfile
	modified:   php/php-suhosin/Portfile
	modified:   php/php-svm/Portfile
	modified:   php/php-svn/Portfile
	modified:   php/php-swoole/Portfile
	modified:   php/php-syck/Portfile
	modified:   php/php-taint/Portfile
	modified:   php/php-text-template/Portfile
	modified:   php/php-timer/Portfile
	modified:   php/php-timezonedb/Portfile
	modified:   php/php-token-stream/Portfile
	modified:   php/php-unit-mock-objects/Portfile
	modified:   php/php-unit/Portfile
	modified:   php/php-uploadprogress/Portfile
	modified:   php/php-uuid/Portfile
	modified:   php/php-vld/Portfile
	modified:   php/php-xcache/Portfile
	modified:   php/php-xdebug/Portfile
	modified:   php/php-xhprof/Portfile
	modified:   php/php-xmlrpc/Portfile
	modified:   php/php-xrange/Portfile
	modified:   php/php-xslcache/Portfile
	modified:   php/php-yaf/Portfile
	modified:   php/php-yaml/Portfile
	modified:   php/php-yaz/Portfile
	modified:   php/php-zip/Portfile
	modified:   php/php-zstd/Portfile
	modified:   php/php5-simpletest/Portfile
	modified:   php/php5-unit-db/Portfile
	modified:   php/php5-unit-selenium/Portfile
	modified:   php/phpsh/Portfile

BjarneDM committed:
php-sqlsrv: update to 5.13.0 for php >= 8.3

BjarneDM committed:
php-Twig: PortGroup obsolete 1.0

BjarneDM committed:
php-Judy: update to 2.4.0 ; php.branches revision

So it seems as if php-Judy has been given a new life
1.0.2 stable 2013-11-03
2.0.0 stable 2025-08-25 php >= 8.0

BjarneDM committed:
php-openswoole: update to 26.2.0 >= php83, 25.2.0 >= php82, 22.1.2 >= php81, 4.12.1

incorporates work from https://github.com/macports/macports-ports/pull/30146

BjarneDMat committed:
php-xmlrpc: libxml2 2.13.9_0 < 2.15.2_0

BjarneDMat commented:
@jmroot Suggested RoadMap : 

1. [PR 31348](https://github.com/macports/macports-ports/pull/31348) php: set BjarneDMat as maintainer. Quite frankly : I won't do any work at all splitting this PR into individual, manageble parts ontill that PR has been merged
2. `update to 8.1.34, 8.2.30, 8.3.30, 8.4.19, 8.5.4 ` squashing a lot of the early stuff. This will be a pre-requisite for all the php extensions due to the changes in the `PortGroup php 1.1`
3. `php-openssl`: fix for Dependency path specified multiple times in depends_build
4. `php-pspell `: for `php >= 8.1` moved to separate Portfile
5. `php-imap`: for `php >= 8.3` moved to separate Portfile
6. `php-mysql ` : update `mariadb` based variants; remove obsolete variants
7. `php-postgresql `: add variants `postgresql{17,18}`
8. `PortGroup obsolete 1.0` - all of them in one go
9. then : update each php extension pr/ PR adding me as (co-)maintainer along the way; a lot of them don't have any maintainer at present. Presently, I've **updated 34 individual extensions**, that now have to be separaded out into individual PRs 😱🤓 And I very much appreciate the help I've received from the MacPorts community on some of these 👍🏻
10. add me as (co-)maintainer to the rest of the php extensions in one go

- When {1} has been merged, I'll do {2}
- When {2} has been merged I'll do {3..8}. {3..7} finishes the work in `lang/php/Portfile`.
- Then {9}. I ought to be able to script those, so I don't have to do too much manual work 🤔😉
- As {10} already is in this PR, then when I've rebased {1..9}, {10} ought to be left over.

Presenty, there're some `libxml2` issues for `php{74,80}` that I'm looking at. They were officially fixed by php for `php82+` and officially backported to `php81`. Patches exist for `php{74,80}`, but there're some issues w/ them I haven't yet been able to fully resolve. `php{74,80}` did build & install from source before the recent massive `libxml2` update.

In other words : building **WILL FAIL** presently for ‼️everything‼️ `< php81` 👎🏻 I'm looking into the `php{74,80}` `libxml2` issues w/ no fixed timeline ⏳, **BUT** won't do anything at all to resolve `< php74` unless specifically asked.

I do think this is a fair RoadMap for both me and MacPorts.
Constructive, concrete comments on, and constructive, concrete suggestions to this RoadMap are more than welcome and encouraged from everyone in the MacPorts community 🤓😍👍🏻

BjarneDMat commented:
@jmroot Please see : 
 #31537 ; #31538 ; #31539 ; #31540 ; #31541

BjarneDMat commented:
> It would be good to set `known_fail yes` in all the php versions that currently won't build with libxml2 2.15, so CI doesn't waste time trying to build them.

I'll work on that now ...
